compiler/variables.ml

```diff
@@ -62,9 +62,9 @@ let init_value_block oc blk numvals =
     for i = 0 to numvals - 1 do
       fprintf oc "%s[%d] = " blk i
     done;
-    fprintf oc "0;\n"
+    fprintf oc "Val_unit;\n"
   end else begin
-    iprintf oc "memset(%s, 0, %d * sizeof(value));\n" blk numvals
+    iprintf oc "memset(%s, Val_unit, %d * sizeof(value));\n" blk numvals
```

Comment:
This use of `memset`  (using byte-sized stores to initialize an array of words) was borderline before, but gets even stranger here.  I'll probably replace it by an explicit initialization loop in a follow-up commit.